docs/docs/how_to/pebblo_retrieval_qa.ipynb

```diff
@@ -0,0 +1,488 @@
+    "        \"eng-support\",\n",
+    "    ],\n",
+    "}\n",
+    "resp = ask(prompt, auth)\n",
+    "print(f\"Question: {question}\\n\\nAnswer: {resp['result']}\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4306cab3-d070-405f-a23b-5c6011a61c50",
+   "metadata": {},
+   "source": [
+    "## Retrieval with Semantic Enforcement"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1c3757cf-832f-483e-aafe-cb09b5130ec0",
+   "metadata": {},
+   "source": [
+    "The PebbloRetrievalQA chain uses SafeRetrieval to ensure that the snippets used in context are retrieved only from documents that comply with the\n",
+    "provided semantic context.\n",
+    "To achieve this, the Gen-AI application must provide a semantic context for this retrieval chain.\n",
+    "This `semantic_context` should include the topics and entities that should be denied for the user accessing the Gen-AI app.\n",
+    "\n",
+    "Below is a sample code for PebbloRetrievalQA with `topics_to_deny` and `entities_to_deny`. These are passed in `semantic_context` to the chain input."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "daf37bf7-9a16-4102-8893-5b698cae1b07",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import List, Optional\n",
+    "\n",
+    "from langchain_community.chains import PebbloRetrievalQA\n",
+    "from langchain_community.chains.pebblo_retrieval.models import (\n",
+    "    ChainInput,\n",
+    "    SemanticContext,\n",
+    ")\n",
+    "\n",
+    "# Initialize PebbloRetrievalQA chain\n",
+    "qa_chain = PebbloRetrievalQA.from_chain_type(\n",
+    "    llm=llm,\n",
+    "    app_name=\"pebblo-semantic-retriever-app\",\n",
+    "    retriever=vectordb.as_retriever(),\n",
+    "    verbose=True,\n",
+    ")\n",
+    "\n",
+    "\n",
+    "def ask(\n",
+    "    question: str,\n",
+    "    topics_to_deny: Optional[List[str]] = None,\n",
+    "    entities_to_deny: Optional[List[str]] = None,\n",
+    "):\n",
+    "    \"\"\"\n",
+    "    Ask a question to the PebbloRetrievalQA chain\n",
+    "    \"\"\"\n",
+    "    semantic_context = dict()\n",
+    "    if topics_to_deny:\n",
+    "        semantic_context[\"pebblo_semantic_topics\"] = {\"deny\": topics_to_deny}\n",
+    "    if entities_to_deny:\n",
+    "        semantic_context[\"pebblo_semantic_entities\"] = {\"deny\": entities_to_deny}\n",
+    "\n",
+    "    semantic_context_obj = (\n",
+    "        SemanticContext(**semantic_context) if semantic_context else None\n",
+    "    )\n",
+    "    chain_input_obj = ChainInput(query=question, semantic_context=semantic_context_obj)\n",
+    "    return qa_chain.invoke(chain_input_obj.dict())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9718819b-f5cd-4212-9947-d18cd507c8b7",
+   "metadata": {},
+   "source": [
+    "### 1. Without semantic enforcement\n",
+    "\n",
+    "Since no semantic enforcement is applied, the system should return the answer.\n"
```

Comment:
```suggestion
    "Since no semantic enforcement is applied, the system should return the answer without excluding any context due to semantic labels associated with the context..\n"
```